OpenEdXMobile/src/main/java/org/edx/mobile/view/MyCoursesListFragment.kt

```diff
@@ -124,23 +127,20 @@ class MyCoursesListFragment : OfflineSupportBaseFragment(), RefreshListener {
             binding.loadingIndicator.root.visibility = View.VISIBLE
             errorNotification.hideError()
         }
-        if (isCached) {
-            getUserEnrolledCourses(courseAPI.enrolledCoursesFromCache, isCached = true)
-        } else {
-            getUserEnrolledCourses(courseAPI.enrolledCourses)
-        }
+        enrolledCoursesApi = if (isCached) courseAPI.enrolledCoursesFromCache else courseAPI.enrolledCourses
+        getUserEnrolledCourses(isCached)
     }
 
-    private fun getUserEnrolledCourses(enrolledCoursesApi: Call<List<EnrolledCoursesResponse>>, isCached: Boolean = false) {
+    private fun getUserEnrolledCourses(isCached: Boolean = false) {
         enrolledCoursesApi.enqueue(object : Callback<List<EnrolledCoursesResponse>> {
             override fun onResponse(call: Call<List<EnrolledCoursesResponse>>, response: Response<List<EnrolledCoursesResponse>>) {
                 if (response.isSuccessful && response.code() == HttpStatus.OK) {
                     populateCourseData(ArrayList(response.body()), isCachedData = isCached)
                     if (isCached) {
-                        getUserEnrolledCourses(courseAPI.enrolledCourses)
+                        loadData(showProgress = response.body()?.isEmpty() == true, isCached = false)
```

Comment:
Should add a comment here:
// Fetch latest data from the server in the background after displaying previously cached data

Reply:
Done